crates/apub/src/objects/post.rs

```diff
@@ -184,7 +184,7 @@ impl Object for ApubPost {
   async fn from_json(page: Page, context: &Data<Self::DataType>) -> LemmyResult<ApubPost> {
     let creator = page.creator()?.dereference(context).await?;
     let community = page.community(context).await?;
-    if community.posting_restricted_to_mods {
+    if community.local && community.posting_restricted_to_mods {
```

Comment:
This commit also moves `moderators.dereference` inside of spawn_try_task in crates/apub/src/objects/community.rs. This way the main action is a bit faster.